guides/theming.md

```diff
@@ -1,311 +1,384 @@
+## What is theming?
 
-Angular Material's theming system enables you to customize components to
-better reflect your product's brand. A theme consists of configurations for the individual
-`color` and `typography` systems in Angular Material. The library's approach to theming reflects
-the guidance from the [Material Design spec][1].
+Angular Material's theming system lets you customize color and typography styles for components
+in your application. The theming system is based on Google's
+[Material Design][material-design-theming] specification.
 
-In Angular Material, you create a color configuration by composing multiple palettes. In
-particular, a color configuration consists of:
+This document describes the concepts and APIs for customizing colors. For typography customization,
+see [Angular Material Typograpy][mat-typography]. For guidance on building components to be
+customizable with this system, see [Theming your own components][theme-your-own].
 
-* A **primary** palette: colors most widely used across all screens and components.
-* An **accent** palette: colors used for the floating action button and interactive elements.
-* A **warn** palette: colors used to convey error state.
-* A **foreground** palette: colors for text and icons.
-* A **background** palette: colors used for element backgrounds.
+[material-design-theming]: https://material.io/design/material-theming/overview.html
+[mat-typography]: https://material.angular.io/guide/typography
+[theme-your-own]: https://material.angular.io/guide/theming-your-components
 
-Additionally, in Angular Material, a configuration may optionally include `typography` settings.
-More information on how typography works can be [found in a dedicated guide][3].
+### Sass
 
-Angular Material theme styles are generated _statically_ at build-time so that your
-app doesn't have to spend cycles generating theme styles when bootstrapping.
+Angular Material's theming APIs are built with [Sass](https://sass-lang.com). This document assumes
+familiary with CSS and Sass basics, including variables, functions, and mixins.
 
-## Using a pre-built theme
+You can use Angular Material without Sass by using a pre-built theme, described in
+[Using a pre-built theme](#using-a-pre-built-theme) below. However, using the library's Sass API
+directly gives you the most control over the styles in your application.
 
-Angular Material comes prepackaged with several pre-built theme css files. These theme files also
-include all the styles for the core (styles common to all components), so you only have to include a
-single css file for Angular Material in your app.
+## Palettes
 
-You can include a theme file directly into your application from
-`@angular/material/prebuilt-themes`
+A **palette** is a collection of colors representing a portion of color space. Each value in this
+collection is called a **hue**. In Material Design, the hues in a palette are numbered from zero
+to 900, lightest to darkest.
```

Comment:
0-900 sounds like every integer, would it make sense to say `the hues in a palette are numbered [0, 100, 200, ..., 900]`?

This may be unnecessary since its obvious from the following code example.

Reply:
May be worth linking this page: https://material.io/design/color/the-color-system.html#color-usage-and-palettes

Reply:
Reworded to be more explicit. It's unfortunate that there's no a nice way to say "each 100 value between 100 and 900 and also 50 for some reason". Also added a link.